jstz_api/tests/wpt.rs

```diff
@@ -0,0 +1,326 @@
+        let result: TestsResult = args.get_or_undefined(1).try_js_into(context)?;
+
+        report.set_harness_result(result)?;
+
+        Ok(JsValue::undefined())
+    }
+}
+
+impl jstz_core::Api for TestHarnessReportApi {
+    fn init(self, context: &mut Context<'_>) {
+        let test_result_callback = FunctionObjectBuilder::new(
+            context.realm(),
+            NativeFunction::from_fn_ptr(Self::test_result_callback),
+        )
+        .name("test_result_callback")
+        .length(1)
+        .build();
+
+        let test_completion_callback = FunctionObjectBuilder::new(
+            context.realm(),
+            NativeFunction::from_fn_ptr(Self::test_completion_callback),
+        )
+        .name("test_completion_callback")
+        .length(3)
+        .build();
+
+        // TODO: Make into a function
+        macro_rules! call_global_function {
+            ($name:literal, $args:expr) => {
+                let value = context
+                    .global_object()
+                    .get(js_string!($name), context)
+                    .expect(&format!("globalThis.{} is undefined", $name));
+
+                let function = value
+                    .as_callable()
+                    .expect(&format!("globalThis.{} is not callable", $name));
+
+                function
+                    .call(&JsValue::undefined(), $args, context)
+                    .expect(&format!("Failed to call globalThis.{}", $name));
+            };
+        }
+
+        call_global_function!("add_result_callback", &[test_result_callback.into()]);
+        call_global_function!(
+            "add_completion_callback",
+            &[test_completion_callback.into()]
+        );
+    }
+}
+
+pub fn run_wpt_test_harness(bundle: &Bundle) -> JsResult<Box<TestHarnessReport>> {
+    let mut rt = Runtime::new()?;
+
+    // Initialize the host-defined object with the test harness report
+    {
+        host_defined!(&mut rt, mut host_defined);
+        host_defined.insert(TestHarnessReport::default());
+    }
+
+    // Define self
+    rt.global_object().insert_property(
+        js_string!("self"),
+        PropertyDescriptor::builder()
+            .value(rt.global_object().clone())
+            .configurable(true)
+            .writable(true)
+            .enumerable(true)
+            .build(),
+    );
+
+    // Run the bundle, evaluating each script in order
+    // Instead of loading the TestHarnessReport script, we initialize it manually
+    for item in &bundle.items {
+        match item {
+            BundleItem::TestHarnessReport => {
+                TestHarnessReportApi.init(&mut rt.context());
+            }
+            BundleItem::Inline(script) | BundleItem::Resource(_, script) => {
+                rt.context().eval(Source::from_bytes(&format!(
+                    "(function() {{ {} }})()",
+                    script
+                )))?;
+
+                println!("done")
+            }
+        }
+    }
```

Comment:
I reported a problem on slack [here](https://tezos-dev.slack.com/archives/C061SSDBN69/p1700144220479869) . 
 
I am not sure at all... but in the `Resource` case, shouldn't it be ?
```
 rt.context().eval(Source::from_bytes(script))?;
 ``` 
  
 I did that on my branch and not only it managed to define `createBuffer`, it also added more tests to the list.

Reply:
I'll investigate this more. My intention was to avoid global bindings accidentally leaking, but perhaps my implementation was incorrect.

Reply:
I decided that your suggestion is a better fit for bundling